tests/command_line/test_dlsnxs2cbf.py

```diff
@@ -9,70 +9,87 @@
 
-@pytest.mark.xfail(reason="Broken for old data while collecting new data")
+def rationalise_whitespace(string: str):
+    """
+    Reformat argparse output in a common layout.
+
+    argparse.parser.print_help() merges consecutive spaces and changes the linebreaks in
+    argparse.parser.description, allowing breaks on a hyphen.  In order to compare
+    expected and actual help output, rationalise a string to a common layout by
+    replacing all instances of '-\n' with '-', and then replacing multiple consecutive
+    whitespace characters with a single space.
+    """
+    return " ".join(string.replace("-\n", "-").split())
+
+
 def test_dlsnxs2cbf(dials_data, tmp_path, capsys):
-    screen = dials_data("thaumatin_eiger_screen", pathlib=True)
-    master = screen / "Therm_6_1_master.h5"
-    run([str(master), "junk_%04d.cbf"])
+    """Test basic behaviour of dxtbx.dlsnxs2cbf."""
+    screen = dials_data("four_circle_eiger", pathlib=True)
+    master = screen / "03_CuHF2pyz2PF6b_P_O" / "CuHF2pyz2PF6b_P_O_02.nxs"
+    run([str(master), "-o", str(tmp_path)])
 
-    output_files = ["junk_%04d.cbf" % j for j in (1, 2, 3)]
+    output_files = [tmp_path / f"{master.stem}_{j:03d}.cbf" for j in (1, 2, 3)]
     captured = capsys.readouterr()
-    for file in output_files:
-        assert file.encode("latin-1") in captured.out
-        assert tmp_path.joinpath(file).is_file()
+    assert f"{master.stem}_###.cbf" in captured.out
+    assert all(path.is_file() for path in output_files)
 
-    expts = ExperimentListFactory.from_filenames(
-        str(tmp_path / file) for file in output_files
-    )
+    expts = ExperimentListFactory.from_filenames(output_files)
     assert all(
-        imgset.get_format_class() == FormatCBFMiniEigerDLS16MSN160
-        for imgset in expts.imagesets()
+        imgset.get_format_class() == FormatCBFMiniEiger for imgset in expts.imagesets()
     )
 
     with h5py.File(master) as fh:
         for i, imgset in enumerate(expts.imagesets()):
             original = fh["/entry/data/data_000001"][i][()]
             sel = np.where(original < original.max())
             np.testing.assert_equal(
-                fh["/entry/data/data_000001"][i][sel],
-                imgset.get_raw_data(0)[0].as_numpy_array()[sel],
+                original[sel], imgset.get_raw_data(0)[0].as_numpy_array()[sel]
             )
 
 
-@pytest.mark.xfail(reason="The dependency chain is broken")
-@pytest.mark.parametrize("remove_axis", ["phi", "chi"], ids=["no phi", "no chi"])
+@pytest.mark.parametrize("remove_axis", ["phi", "kappa"], ids=["no phi", "no kappa"])
 def test_dlsnxs2cbf_deleted_axis(dials_data, tmp_path, remove_axis):
-    """Check that a master file without φ or χ axes processes OK."""
-    screen = dials_data("thaumatin_eiger_screen", pathlib=True)
-    master = "Therm_6_1_master.h5"
+    """Check that a master file without φ or κ axes processes OK."""
+    screen = dials_data("four_circle_eiger", pathlib=True) / "03_CuHF2pyz2PF6b_P_O"
```

Comment:
This increases the test download requirements by 1.2 GB (the thaumatin eiger screen data is still used elsewhere). Sure this is a good idea?

For comparison: the entire build container is limited to 10 GB.

Reply:
Ah, OK, I didn't realise we had that limitation.  I'll go back to the drawing board and try to collect a smaller data set.  In the meantime, I'll mark this as a draft.

Reply:
At Diamond we can keep the data on a network drive and just reference it there, but tests on Azure always need to download all the test data _somehow_. Whether that's via a direct download, a build cache, a build artefact, or via another route.